src/policies/EmissionManager.sol

```diff
@@ -0,0 +1,469 @@
+}
+
+interface Clearinghouse {
+    function principalReceivables() external view returns (uint256);
+}
+
+contract EmissionManager is Policy, RolesConsumer {
+    using FullMath for uint256;
+
+    // ========== ERRORS ========== //
+
+    error OnlyTeller();
+    error InvalidMarket();
+    error InvalidCallback();
+    error InvalidParam(string parameter);
+    error CannotRestartYet(uint48 availableAt);
+    error RestartTimeframePassed();
+    error NotActive();
+    error AlreadyActive();
+
+    // ========== EVENTS ========== //
+
+    event SaleCreated(uint256 marketID, uint256 saleAmount);
+    event BackingUpdated(uint256 newBacking, uint256 supplyAdded, uint256 reservesAdded);
+
+    // ========== DATA STRUCTURES ========== //
+
+    struct BaseRateChange {
+        uint256 changeBy;
+        uint48 beatsLeft;
+        bool addition;
+    }
+
+    // ========== STATE VARIABLES ========== //
+
+    /// @notice active base emissions rate change information
+    /// @dev active until beatsLeft is 0
+    BaseRateChange public rateChange;
+
+    // Modules
+    TRSRYv1 public TRSRY;
+    PRICEv1 public PRICE;
+    MINTRv1 public MINTR;
+    CHREGv1 public CHREG;
+
+    // Tokens
+    // solhint-disable const-name-snakecase
+    ERC20 public immutable ohm;
+    IgOHM public immutable gohm;
+    ERC20 public immutable reserve;
+    ERC4626 public immutable sReserve;
+
+    // External contracts
+    IBondSDA public auctioneer;
+    address public teller;
+
+    // Manager variables
+    uint256 public baseEmissionRate;
+    uint256 public minimumPremium;
+    uint48 public vestingPeriod; // initialized at 0
+    uint256 public backing;
+    uint8 public beatCounter;
+    bool public locallyActive;
+    uint256 public activeMarketId;
+
+    uint8 internal _oracleDecimals;
+    uint8 internal immutable _ohmDecimals;
+    uint8 internal immutable _reserveDecimals;
+
+    /// @notice timestamp of last shutdown
+    uint48 public shutdownTimestamp;
+    /// @notice time in seconds that the manager needs to be restarted after a shutdown, otherwise it must be re-initialized
+    uint48 public restartTimeframe;
+
+    uint256 internal constant ONE_HUNDRED_PERCENT = 1e18;
+
+    // ========== SETUP ========== //
+
+    constructor(
+        Kernel kernel_,
+        address ohm_,
+        address gohm_,
+        address reserve_,
+        address sReserve_,
+        address auctioneer_,
+        address teller_
+    ) Policy(kernel_) {
+        // Set immutable variables
+        if (ohm_ == address(0)) revert("OHM address cannot be 0");
+        if (gohm_ == address(0)) revert("gOHM address cannot be 0");
+        if (reserve_ == address(0)) revert("DAI address cannot be 0");
+        if (sReserve_ == address(0)) revert("sDAI address cannot be 0");
+        if (auctioneer_ == address(0)) revert("Auctioneer address cannot be 0");
+
+        ohm = ERC20(ohm_);
+        gohm = IgOHM(gohm_);
+        reserve = ERC20(reserve_);
```

Comment:
you should flip this around. Set sReserve, then grab the reserve from the sReserve.

sReserve.asset() should get it to you iirc. Refer to 4626